components/calendar/src/hebrew.rs

```diff
@@ -0,0 +1,848 @@
+//!
+//! // `DateTime` checks
+//! assert_eq!(hebrew_datetime.date.year().number, 3425);
+//! assert_eq!(hebrew_datetime.date.month().ordinal, 10);
+//! assert_eq!(hebrew_datetime.date.day_of_month().0, 11);
+//! assert_eq!(hebrew_datetime.time.hour.number(), 13);
+//! assert_eq!(hebrew_datetime.time.minute.number(), 1);
+//! assert_eq!(hebrew_datetime.time.second.number(), 0);
+//! ```
+
+use crate::calendar_arithmetic::{ArithmeticDate, CalendarArithmetic};
+use crate::helpers::{self, div_rem_euclid, div_rem_euclid_f64, final_func, next_u8};
+use crate::julian::Julian;
+use crate::rata_die::RataDie;
+use crate::types::Moment;
+use crate::Iso;
+use crate::{types, Calendar, CalendarError, Date, DateDuration, DateDurationUnit, DateTime};
+use ::tinystr::tinystr;
+
+/// Hebrew Arithmetical Calendar
+#[derive(Copy, Clone, Debug, Default, Hash, Eq, PartialEq, PartialOrd, Ord)]
+#[allow(clippy::exhaustive_structs)]
+pub struct Hebrew;
+
+// Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2206
+const FIXED_HEBREW_EPOCH: RataDie = Julian::fixed_from_julian_integers(-3761, 10, 8);
+
+// The Hebrew Months
+const NISAN: u8 = 1;
+const IYYAR: u8 = 2;
+#[allow(dead_code)]
+const SIVAN: u8 = 3;
+const TAMMUZ: u8 = 4;
+#[allow(dead_code)]
+const AV: u8 = 5;
+const ELUL: u8 = 6;
+const TISHRI: u8 = 7;
+const MARHESHVAN: u8 = 8;
+const KISLEV: u8 = 9;
+const TEVET: u8 = 10;
+#[allow(dead_code)]
+const SHEVAT: u8 = 11;
+const ADAR: u8 = 12;
+const ADARII: u8 = 13;
+
+/// The inner date type used for representing [`Date`]s of [`Hebrew`]. See [`Date`] and [`Hebrew`] for more details.
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
+pub struct HebrewDateInner(ArithmeticDate<Hebrew>);
+
+impl CalendarArithmetic for Hebrew {
+    fn month_days(year: i32, month: u8) -> u8 {
+        Self::last_day_of_hebrew_month(year, month)
+    }
+
+    fn months_for_every_year(year: i32) -> u8 {
+        Self::last_month_of_hebrew_year(year)
+    }
+
+    fn days_in_provided_year(year: i32) -> u16 {
+        Self::days_in_hebrew_year(year)
+    }
+
+    fn is_leap_year(year: i32) -> bool {
+        div_rem_euclid(7 * year + 1, 19).1 < 7
+    }
+
+    fn last_month_day_in_year(year: i32) -> (u8, u8) {
+        let month = Self::last_month_of_hebrew_year(year);
+        let day = Self::last_day_of_hebrew_month(year, month);
+
+        (month, day)
+    }
+}
+
+impl Calendar for Hebrew {
+    type DateInner = HebrewDateInner;
+
+    fn date_from_codes(
+        &self,
+        era: types::Era,
+        year: i32,
+        month_code: types::MonthCode,
+        day: u8,
+    ) -> Result<Self::DateInner, CalendarError> {
+        let year = if era.0 == tinystr!(16, "hebrew") {
+            year
+        } else {
+            return Err(CalendarError::UnknownEra(era.0, self.debug_name()));
+        };
+
+        ArithmeticDate::new_from_codes(self, year, month_code, day).map(HebrewDateInner)
```

Comment:
You need to handle the leap month `M05L`, both here and in the `.month()` function.

Reply:
Issue: you need to handle M05L on input, too